src/utils/mtev_watchdog.c

```diff
@@ -417,6 +419,24 @@ void mtev_self_diagnose(int sig, siginfo_t *si, void *uc) {
   raise(sig);
 }
 
+char *external_diagnose = NULL;
+void mtev_external_diagnose(int sig, siginfo_t *si, void *uc) {
+  (void)si;
+  (void)uc;
+  pid_t pid = syscall(SYS_gettid);
```

Comment:
This looks like Linux only. There should be an alternative implementation that uses getpid()

Reply:
yes you are right, fixed.